src/hipoc/hipoc_program.cpp

```diff
@@ -369,6 +369,8 @@ std::string HIPOCProgram::GetCodeObjectBlob() const
     return {impl->binary.data(), impl->binary.size()};
 }
 
+void HIPOCProgram::DelTemp() const { impl->dir = boost::none; }
```

Comment:
Please also invalidate `impl->hsaco_file` (just clear it).

Reply:
When `impl->hsaco_file` is invalid, `HIPOCProgram::GetCodeObjectPathname()` must `MIOPEN_THROW(miopenStatusInternalError...)`.

Reply:
Please add the following to the documentation of `GetCodeObjectPathname():
```C++
/// This function should not be called after FreeCodeObjectFileStorage().
```